contracts/eth/nearbridge/contracts/AdminControlled.sol

```diff
@@ -34,22 +48,18 @@ contract AdminControlled {
         uint key,
         uint value,
         uint mask
-    ) public onlyAdmin {
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
         assembly {
             let oldval := sload(key)
             sstore(key, xor(and(xor(value, oldval), mask), oldval))
         }
     }
 
-    function adminSendEth(address payable destination, uint amount) public onlyAdmin {
+    function adminSendEth(address payable destination, uint amount) external onlyRole(DEFAULT_ADMIN_ROLE) {
         destination.transfer(amount);
     }
 
-    function adminReceiveEth() public payable onlyAdmin {}
+    function adminReceiveEth() external payable {}
 
-    function adminDelegatecall(address target, bytes memory data) public payable onlyAdmin returns (bytes memory) {
-        (bool success, bytes memory rdata) = target.delegatecall(data);
-        require(success);
-        return rdata;
-    }
```

Comment:
It works for contracts, but as we are running this as a proxy there are some security reason to avoid using a delegate call inside the implementation.
for more details see [here](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#potentially-unsafe-operations)